### PR TITLE
Add support for "netbox [ZONES...] syntax" in Corefile

### PR DESCRIPTION
**Pull request by andrewheberle**
This PR makes the plugin support the same syntax as many other CoreDNS plugins as follows:

```
. {
  netbox example.org {
    ...
  }
}
```

This PR should allow #17 to be closed.

The original syntax still works, but this then sets the plugin to respond to whatever zones are in the main Corefile block, with the following handling all zones:

```
. {
  netbox {
    ...
  }
}
```
And this example only handles `example.org`;

```
example.org {
  netbox {
    ...
  }
}
```

As part of this change the config parsing was made to be more similar to a few of the included CoreDNS plugins.

The rest of the changes are README updates and test changes due to moving config parsing to its own function in `netbox.parseNetbox`.

### CODE REVIEW

**Comment by oz123** [comment]
@andrewheberle Really, really nice work.
Can you please update `Corefile.example` and add some comments in it?

File: Corefile.example [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
-   }
+  hosts {
+    10.0.0.1 example.org
+	  10.0.0.3 definitely-not-in-netbox.org
+    fallthrough
+  }
 }
+
+# The following examples specify what zone(s) should be handled by netbox:
+#
```
Comment (oz123) [inline Corefile.example:17]: What is the difference here between the zones?

File: Corefile.example [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+    token 0123456789abcdef0123456789abcdef01234567
+    url http://localhost:8000/api/ipam/ip-addresses
+    localCacheDuration 1h
+  }
 
-   hosts {
-        10.0.0.1 example.org
-	10.0.0.3 definitely-not-in-netbox.org
-        fallthrough
-   }
```
Comment (oz123) [inline Corefile.example:13]: So, now hosts which are not found in netbox will be looked in the inline hosts section?

File: Corefile.example [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
+  hosts {
+    10.0.0.1 example.org
+	  10.0.0.3 definitely-not-in-netbox.org
```
Comment (oz123) [inline Corefile.example:11]: I think the indentation with tab can be removed here.

**Review by oz123 — COMMENTED** [review]
The examples are looking good already. I think clarifying the questions in the examples would be great.

File: Corefile.example (hunk 2 above)
Comment (andrewheberle) [reply Corefile.example:13]: I have added some comments relating to this, but in a word? No.

This all depends on the ordering of the plugins in `plugins.cfg`.

If *netbox* was **before** *hosts* in `plugins.cfg` then *netbox* will handle all queries and never pass the query to *hosts* as there is no `fallthrough` directive in the *netbox* config block.

If *netbox* was **after** *hosts* in `plugins.cfg` then *hosts* will handle the query first and then pass the request to *netbox* if it cannot handle it due to the `fallthrough` directive in the *hosts* config block.

File: Corefile.example (hunk 1 above)
Comment (andrewheberle) [reply Corefile.example:30]: Hopefully the added comments make this clearer also.

File: Corefile.example (hunk 3 above)
Comment (andrewheberle) [reply Corefile.example]: Fixed.

**Review by oz123 — APPROVED** [review]

**Pull request merged**